### PR TITLE
[Security Content] Tags Reform

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

This PR introduces the following categories to be used with the tags:

- Application
- Data Source
- Domain
- OS
- Resources
- Rule Type
- Tactic
- Threat
- Use Case

<details>
  <summary>Expand me to see the tags!</summary>

```
'Use Case: Threat Detection', 582
'Domain: Endpoint', 537
'OS: Windows', 364
'Data Source: Elastic Endgame', 257
'Resources: Investigation Guide', 223
'Domain: Cloud', 173
'Tactic: Defense Evasion', 155
'Tactic: Persistence', 123
'OS: Linux', 117
'Use Case: Identity and Access Audit', 88
'Tactic: Credential Access', 86
'OS: macOS', 85
'Tactic: Execution', 79
'Tactic: Privilege Escalation', 75
'Data Source: AWS', 60
'Data Source: Amazon Web Services', 60
'Rule Type: Machine Learning', 47
'Rule Type: ML', 47
'Tactic: Lateral Movement', 43
'Use Case: Configuration Audit', 41
'Tactic: Command and Control', 40
'Data Source: Azure', 37
'Tactic: Discovery', 36
'Tactic: Initial Access', 35
'Data Source: Microsoft 365', 31
'Data Source: Okta', 29
'Use Case: Log Auditing', 24
'Use Case: Vulnerability', 23
'Data Source: Google Cloud Platform', 23
'Data Source: GCP', 23
'Data Source: Google Workspace', 23
'Tactic: Impact', 21
'Data Source: PowerShell Logs', 21
'Data Source: Active Directory', 20
'Use Case: Active Directory Monitoring', 20
'Use Case: Network Security Monitoring', 15
'Use Case: Asset Visibility', 15
'Tactic: Collection', 13
'Data Source: Elastic Defend for Containers', 12
'Domain: Container', 12
'Data Source: Kubernetes', 12
'Data Source: Sysmon Only', 11
'Data Source: APM', 5
'Threat: BPFDoor', 3
'Rule Type: Higher-Order Rule', 2
'Tactic: Exfiltration', 2
'Data Source: CyberArk PAS', 2
'Threat: Lightning Framework', 2
'Threat: Orbit', 2
'Use case: Guided Onboarding', 1
'Data Source: Zoom', 1
'Tactic: Reconnaissance', 1
'Data Source: Elastic Defend', 1
'Threat: TripleCross', 1
'Threat: Rootkit', 1
'Tactic: Resource Development', 1
'Threat: Cobalt Strike', 1)]
```
<\details>

Date was changed to "2023/00/00" so I can bump them all at once if needed

### CODE REVIEW

**Comment by terrancedejesus** [comment]
Thanks for starting this Jon! Just a reminder to anyone, updating tags bumps rule versions.

I like the categorization you made, I think it makes sense and is concise. We will need to update the unit tests to enforce these if we move forward. Speaking of which, when /if we merge and backport, some branches will be in a failing state, but these can be fixed manually if we identify forked rules ahead of the merge.

Programmatically, while the search UI doesn't have filters for these categories, I would have hopes it is on the roadmap. Most of these tags can be programmatically determined and put into the API formatted rules that go into prebuilt rules package, similar to build time fields, thus upstream the user would have more flexibility on what to search for and we would be less reliant on tags.

**Comment by approksiu** [comment]
The reviewed list of categories:

- Data Source
- Domain
- OS
- Resources
- Rule Type
- Tactic
- Threat
- Use Case

The reviewed list of tags:

- Data Source: Active Directory
- Data Source: Amazon Web Services
- Data Source: APM
- Data Source: AWS
- Data Source: Azure
- Data Source: CyberArk PAS
- Data Source: Elastic Defend
- Data Source: Elastic Defend for Containers
- Data Source: Elastic Endgame
- Data Source: GCP
- Data Source: Google Cloud Platform
- Data Source: Google Workspace
- Data Source: Kubernetes
- Data Source: Microsoft 365
- Data Source: Okta
- Data Source: PowerShell Logs
- Data Source: Sysmon only
- Data Source: Zoom
- Domain: Application
- Domain: Cloud
- Domain: Endpoint
- Domain: Network
- OS: Linux
- OS: macOS
- OS: Windows
- Resources: Investigation Guide
- Rule Type: Higher-Order Rule
- Rule Type: Machine Learning
- Rule Type: ML
- Tactic: Collection
- Tactic: Command and Control
- Tactic: Credential Access
- Tactic: Defense Evasion
- Tactic: Discovery
- Tactic: Execution
- Tactic: Exfiltration
- Tactic: Impact
- Tactic: Impair Defenses
- Tactic: Initial Access
- Tactic: Lateral Movement
- Tactic: Persistence
- Tactic: Privilege Escalation
- Tactic: Resource Development
- Threat: BPFDoor
- Threat: Cobalt Strike
- Threat: Lightning Framework
- Threat: Orbit
- Threat: Rootkit
- Threat: TripleCross
- Use Case: Asset Visibility
- Use Case: Configuration Audit
- Use case: Guided Onboarding
- Use Case: Identity and Access Audit
- Use Case: Log Auditing
- Use Case: Network Security Monitoring
- Use Case: Threat Detection
- Use Case: Vulnerability

Maybe another tag - discuss if it falls under Endpoint
- Domain: Container

**Comment by imays11** [comment]
I think we should include the `Container` tag. It could technically fall under `Endpoint`,  `Cloud`, or even `Application`. It should be distiguished as a tag of its own, I think people would expect this. 

We don't need `Amazon Web Services` since we have `AWS`. We should also choose between having `Machine Learning` vs `ML`, we don't need both. 

I do agree with Terrance that `Threat Detection` seems like it would be a part of every rule so may not be worth including as a tag.

**Comment by approksiu** [comment]
> I think we should include the `Container` tag. It could technically fall under `Endpoint`, `Cloud`, or even `Application`. It should be distiguished as a tag of its own, I think people would expect this.
> 
> We don't need `Amazon Web Services` since we have `AWS`. We should also choose between having `Machine Learning` vs `ML`, we don't need both.
> 
> I do agree with Terrance that `Threat Detection` seems like it would be a part of every rule so may not be worth including as a tag.

Thanks @imays11! 
We have added ML/Machine Learning  and AWS/Amazon Web Services to accommodate how people search tags.

File: rules/cross-platform/defense_evasion_timestomp_touch.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ integration = ["endpoint"]
 maturity = "production"
 min_stack_comments = "New fields added: required_fields, related_integrations, setup"
 min_stack_version = "8.3.0"
-updated_date = "2022/12/14"
+updated_date = "2023/00/00"
```
Comment (brokensound77) [inline rules/cross-platform/defense_evasion_timestomp_touch.toml:7]: bug in the date conversion on many of these

Comment (brokensound77) [reply rules/cross-platform/defense_evasion_timestomp_touch.toml:7]: nvm, just noticed your comment 🙂

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Review by terrancedejesus — APPROVED** [review]

**Review by eric-forte-elastic — APPROVED** [review]
Looks good to me :+1: 

Still going to approve but just to double check. It is expected that a number of tags are removed in this PR right? I am assuming it is, but putting notes here in case. 

For instance in these rules:

<details><summary>Example Rules with tag deletion</summary>
<p>

- rules/cross-platform/initial_access_zoom_meeting_with_no_passcode.toml
- rules/integrations/aws/credential_access_aws_iam_assume_role_brute_force.toml
- rules/integrations/aws/credential_access_iam_user_addition_to_group.toml
- rules/integrations/aws/credential_access_new_terms_secretsmanager_getsecretvalue.toml
- rules/integrations/aws/exfiltration_ec2_snapshot_change_activity.toml
- rules/integrations/aws/impact_cloudwatch_log_stream_deletion.toml

</p>
</details> 

Tags that were deleted and not directly replaced:
- SecOps
- Data Protection
- Continuous Monitoring 
- Elastic
- Host
- GTFOBins

**Comment by w0rk3r** [comment]
@eric-forte-elastic Yup, we dropped a few tags here, some of them were very specific and didn't fit in any of the categories we came up with, others served a purpose that can be achieved through the UI now (e.g., Elastic), and others were too generic to mean anything in the rule context.

**Pull request merged**